lib/mri/rubygems/requirement.rb

```diff
@@ -4,7 +4,15 @@
 
 # If we're being loaded after yaml was already required, then
 # load our yaml + workarounds now.
-Gem.load_yaml if defined? ::YAML
+if defined? ::YAML
+  # Truffle: this is conditional because at this point #gem cannot be defined (it is
+  # defined after loading rubygems/core_ext/kernel_gem which comes after this file).
+  # In the case of lazy RubyGems, #gem exists but is a stub while loading RubyGems,
+  # which would cause a stack overflow. Removing #gem is not OK for concurrency.
+  unless defined?(gem) and File.basename(method(:gem).source_location[0]) == 'lazy-rubygems.rb'
```

Comment:
I thought about it, changed the line to use `&&` thinking about what you would say, but then it did not survive further refactoring :smile: